scripts/wrist_test.py

```diff
@@ -0,0 +1,245 @@
+                msg.position = [t2]
+                msg.velocity = [0]
+                msg.effort = [0]
+                publisher.publish(msg)
+                rate.sleep()
+
+
+def move_shoulder_elbow_wrist(left_or_right="right", move_elbow=True, move_wrist=True,
+                              shoulder_range=None, elbow_range=None, wrist_range=None):
+    shoulder_range = [0.0, -1.1] if not shoulder_range else shoulder_range
+    elbow_range = [0.0, 0.6] if not elbow_range else elbow_range
+    wrist_range = [-1.5, -0.05] if not wrist_range else wrist_range
+
+    msg = JointState()
+    num_samples = 200
+
+    msg.name = ["shoulder_" + left_or_right + "_axis0"]
+    targets = [np.linspace(shoulder_range[0], shoulder_range[1], num_samples)]
+    if move_elbow:
+        msg.name.append("elbow_" + left_or_right + "_axis0")
+        targets.append(np.linspace(elbow_range[0], elbow_range[1], num_samples))
+    if move_wrist:
+        msg.name.append("wrist_" + left_or_right + "_axis0")
+        targets.append(np.linspace(wrist_range[0], wrist_range[1], num_samples))
+
+    targets = np.array(targets)
+
+    for i in range(num_samples):
+        msg.position = list(targets[:, i])
+        msg.velocity = [0] * len(msg.name)
+        msg.effort = [0] * len(msg.name)
+        publisher.publish(msg)
+        rate.sleep()
+
+
+def move_wrist(left_or_right="right", axis=0, ranges=None):
+    ranges = [-1.5, -0.05] if not ranges else ranges
+
+    msg = JointState()
+    samples = 200
+    msg.name = ["wrist_" + left_or_right + "_axis" + str(axis), "wrist_" + left_or_right + "_axis0"]
+    msg.velocity = [0] * len(msg.name)
+    msg.effort = [0] * len(msg.name)
+
+    for i in range(6):
+
+        if not i % 2:
+            targets = np.linspace(ranges[-1], ranges[0], samples)
+
+            for t in targets:
+                msg.position = [t, t]
+                publisher.publish(msg)
+                rospy.sleep(0.01)
+
+        else:
+            targets = np.linspace(ranges[0], ranges[-1], samples)
+
+            for t in targets:
+                msg.position = [t, t]
+                publisher.publish(msg)
+                rospy.sleep(0.01)
+
+
+def move_all(move_type="sequentially", shoulder_range=None, elbow_range=None, wrist_range=None, head_range=None):
+    shoulder_range = [0.0, -1.1] if not shoulder_range else shoulder_range
+    elbow_range = [0.0, 0.6] if not elbow_range else elbow_range
+    wrist_range = [-1.5, -0.05] if not wrist_range else wrist_range
+    head_range = [-0.56, 0.56] if not head_range else head_range
+
+    msg = JointState()
+    msg.name = ["shoulder_left_axis0", "elbow_left_axis0", "wrist_left_axis0", "wrist_left_axis1",
+                "shoulder_right_axis0", "elbow_right_axis0", "wrist_right_axis0", "wrist_right_axis1",
+                "head_axis0"]
+
+    if move_type == "sequentially":
+        targets = np.linspace(shoulder_range[0], shoulder_range[1], 200)
```

Comment:
200 - the last parameter is hardcoded everyone. it's bad style of coding. use a variable, and as mentioned in my comment above, it must be a parameter passed to the function

Reply:
Done